astra/src/main/java/com/slack/astra/bulkIngestApi/opensearch/BulkApiRequestParser.java

```diff
@@ -104,6 +105,29 @@ public static Trace.Span fromIngestDocument(
     spanBuilder.setTimestamp(
         TimeUnit.MICROSECONDS.convert(timestampInMillis, TimeUnit.MILLISECONDS));
 
+    // Today @timestamp is parsed in getTimestampFromIngestDocument and set as timestampInMillis
+    // However we store it as _timesinceepoch in the trace.proto
+    // Let's also store the original field name and classify it as a date
+    if (ingestDocument.hasField("@timestamp")) {
+      String dateString = ingestDocument.getFieldValue("@timestamp", String.class);
+      LocalDateTime localDateTime =
+          LocalDateTime.parse(dateString, DateTimeFormatter.ISO_DATE_TIME);
+      // As I understand frontend UIs expect this to be constant at UTC and they adjust it based on
```

Comment:
Instant can always be considered UTC (ie, https://www.baeldung.com/java-instant-vs-localdatetime#:~:text=The%20easiest%20way%20to%20think,in%20the%20UTC%20time%20zone). You can see this also in the javadocs for `Instant.atOffset()`.

As such we'd always want to store this at UTC, and you are correct that timezone conversions are done at query time - https://github.com/slackhq/astra/blob/master/astra/src/main/java/com/slack/astra/logstore/opensearch/OpenSearchAdapter.java#L985